cmd/sink/routers/routers.go

```diff
@@ -189,3 +196,44 @@ func (c *Controller) receiveCloudEvent(ctx context.Context, event cloudevents.Ev
 func (c *Controller) CloudEventsHandler(ctx *gin.Context) {
 	c.ceHandler.ServeHTTP(ctx.Writer, ctx.Request)
 }
+
+func (c *Controller) Livez(ctx *gin.Context) {
+	observabilityConfig := os.Getenv(observability.OtelStartEnvVar)
+	if observabilityConfig == "" {
+		observabilityConfig = "disabled"
+	}
+	ctx.JSON(http.StatusOK, gin.H{
+		"Code":          http.StatusOK,
+		"ginMode":       gin.Mode(),
+		"openTelemetry": observabilityConfig,
+		"message":       "healthy",
+	})
+}
+
+// Readyz checks connections to the Database and to the Kubernetes API
+func (c *Controller) Readyz(ctx *gin.Context) {
+	err := c.Db.Ping(ctx.Request.Context())
+	if err != nil {
+		abort.Abort(ctx, err, http.StatusServiceUnavailable)
+		return
+	}
+	ns := os.Getenv(namespaceEnvVar)
+	if ns == "" {
+		err = fmt.Errorf(
+			"Could not determine the KubeArchive namespace. Environment variable %s is not set", namespaceEnvVar,
+		)
+		abort.Abort(ctx, err, http.StatusServiceUnavailable)
+	}
+	cm := corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "ConfigMap",
+		},
+	}
+	resource, _ := meta.UnsafeGuessKindToResource(cm.GroupVersionKind())
+	_, err = c.K8sClient.Resource(resource).Namespace(ns).List(ctx.Request.Context(), metav1.ListOptions{})
+	if err != nil {
+		abort.Abort(ctx, err, http.StatusServiceUnavailable)
+	}
```

Comment:
Why we check that we have permissions to list ConfigMaps? I guess this may change at any time, but wouldn't be easier to check for permissions at startup instead?

Reply:
I guess it kind of is a check on the permissions, but what I'm trying to check is that the sink can successfully connect to the k8s api, so that kubernetes will change the ready status of the sink if it cannot